File: snippets/csharp/BuyBtc.cs
using Breez.Sdk;

public class BuyBtcSnippets
{
    public void FetchOnchainLimits(BindingLiquidSdk sdk)
    {
        // ANCHOR: onchain-limits
        try
        {
            var currentLimits = sdk.FetchOnchainLimits();
            Console.WriteLine($"Minimum amount, in sats: {currentLimits.receive.minSat}");
            Console.WriteLine($"Maximum amount, in sats: {currentLimits.receive.maxSat}");
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: onchain-limits
    }

    public void PrepareBuyBtc(BindingLiquidSdk sdk, OnchainPaymentLimitsResponse currentLimits)
    {
        // ANCHOR: prepare-buy-btc
        try
        {
            var req = new PrepareBuyBitcoinRequest(BuyBitcoinProvider.MOONPAY, currentLimits.receive.minSat);
            var prepareResponse = sdk.PrepareBuyBitcoin(req);

            // Check the fees are acceptable before proceeding
            var receiveFeesSat = prepareResponse.feesSat;
            Console.WriteLine($"Fees: {receiveFeesSat} sats");
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: prepare-buy-btc
    }

    public void BuyBtc(BindingLiquidSdk sdk, PrepareBuyBitcoinResponse prepareResponse)
    {
        // ANCHOR: buy-btc
        try
        {
            var req = new BuyBitcoinRequest(prepareResponse);
            var url = sdk.BuyBitcoin(req);
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: buy-btc
    }
}

File: snippets/python/src/buy_btc.py
import logging
from breez_sdk_liquid import BindingLiquidSdk, OnchainPaymentLimitsResponse, BuyBitcoinProvider, PrepareBuyBitcoinRequest, PrepareBuyBitcoinResponse, BuyBitcoinRequest

def fetch_onchain_limits(sdk: BindingLiquidSdk):
    # ANCHOR: onchain-limits
    try:
        current_limits = sdk.fetch_onchain_limits()
        logging.debug(f"Minimum amount, in sats {current_limits.receive.min_sat}")
        logging.debug(f"Maximum amount, in sats {current_limits.receive.max_sat}")
        return current_limits
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: onchain-limits

def prepare_buy_btc(sdk: BindingLiquidSdk, current_limits: OnchainPaymentLimitsResponse):
    # ANCHOR: prepare-buy-btc
    try:
        req = PrepareBuyBitcoinRequest(BuyBitcoinProvider.MOONPAY, current_limits.receive.min_sat)
        prepare_response = sdk.prepare_buy_bitcoin(req)

        # Check the fees are acceptable before proceeding
        receive_fees_sat = prepare_response.fees_sat
        logging.debug(f"Fees: {receive_fees_sat} sats")
        return prepare_response
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: prepare-buy-btc

def buy_btc(sdk: BindingLiquidSdk, prepareResponse: PrepareBuyBitcoinResponse):
    # ANCHOR: buy-btc
    try:
        req = BuyBitcoinRequest(prepareResponse)
        url = sdk.buy_bitcoin(req)
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: buy-btc
